File: solutions/JavaScript/minimum_size_subarray_sum_sliding_window.js
function minSubArrayLenSlidingWindow(target, nums) {
    let left = 0;
    let currentSum = 0;
    let minLength = Infinity;

    for (let right = 0; right < nums.length; right++) {
        currentSum += nums[right];

        while (currentSum >= target) {
            minLength = Math.min(minLength, right - left + 1);
            currentSum -= nums[left];
            left++;
        }
    }

    return minLength === Infinity ? 0 : minLength;
}

const nums = [2, 3, 1, 2, 4, 3];
const target = 7;
const result = minSubArrayLenSlidingWindow(target, nums);
console.log(result);  // 2

File: solutions/C++/minimum_size_subarray_sum_sliding_window.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int minSubArrayLenSlidingWindow(int target, std::vector<int>& nums) {
    int left = 0;
    int current_sum = 0;
    int min_length = INT_MAX;

    for (int right = 0; right < (int)nums.size(); right++) {
        current_sum += nums[right];

        while (current_sum >= target) {
            min_length = std::min(min_length, right - left + 1);
            current_sum -= nums[left];
            left++;
        }
    }

    return min_length == INT_MAX ? 0 : min_length;
}

int main() {
    std::vector<int> nums = {2, 3, 1, 2, 4, 3};
    int target = 7;
    int result = minSubArrayLenSlidingWindow(target, nums);
    std::cout << result << std::endl;
    return 0;
}
